### DISCUSSION TITLE
✔ Python editor node

### DISCUSSION THREAD

**Discussion by Tokarzewski** [IndiePython/nodezator#25] [Feature Request]
Python editor node that could create or edit nodes within nodezator

**Comment by KennedyRichard** [comment]
Do you mean like group nodes in Blender3D: Being able to create a node by visually combining existing ones?

**Reply by Tokarzewski** [reply]
No, more like the GHPython node in Grasshopper.

**Reply by KennedyRichard** [reply]
Okay, I'll take a look.

**Reply by Tokarzewski** [reply]
https://developer.rhino3d.com/guides/rhinopython/ghpython-component/

**Comment by KennedyRichard** [comment]
Mr. @Tokarzewski,

Thank you for your patience. I'm sorry for taking so long.

While researching other apps for my current task of implementing branching in Nodezator, I took the opportunity to also look into this feature request you made.

The new 1.3.1 version of Nodezator (we are actually at 1.3.2 now) ships with a new node that does exactly what GHPython nodes do in Grasshopper. It is called `namespace_from_exec` and you can find it on the popup menu **Useful encapsulations > namespace_from_exec**.

The only difference is that it returns a dict from which you can retrieve any variable you want from your script. You can easily get any variable you want from this dict by using one of the other default nodes. For instance, here's an example of using the `get_at_string` node (also from the **Useful encapsulations** menu) to get the `c` variable defined inside the script:

![namespace_from_exec demo](https://i.imgur.com/DwVxhRa.png)

If there's anything you'd like to add, I'm all ears.

---

As a curiosity, even before introducing those new nodes Nodezator could already achieve the same result just with default nodes. Here's how one would do it:

![namespace_from_exec hack](https://i.imgur.com/Q1KV5A6.png)

But it is a bit hacky, which is why I created the new node for convenience (as well as the `get_at_string` node and other similar ones).
